src/socket.c

```diff
@@ -0,0 +1,49 @@
+/*
+   Copyright (c) 2017, Christof Schulze <christof.schulze@gmx.net>
+   All rights reserved.
+
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted provided that the following conditions are met:
+
+   1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+   2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+   AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+   IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+   FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+   DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+   SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+   CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+   OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+   */
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include "socket.h"
+#include "error.h"
+
+#define BUFLEN 12
+
+void socket_init(socket_ctx *ctx) {
+	printf("init socket\n");
+	ctx->fd = socket(AF_INET6, SOCK_STREAM | SOCK_NONBLOCK, 0);
+	bind(ctx->fd, (struct sockaddr *)&ctx->socket, sizeof(ctx->socket));
+	listen(ctx->fd, 5);
+}
+
+void socket_handle_out(socket_ctx *ctx, size_t count) {
+	int fd  = accept(ctx->fd, (struct sockaddr*) NULL, NULL);
+
+	char tmp[BUFLEN];
```

Comment:
No need to use a buffer here, just:
```C
dprintf(fd, "%zu", count);
```
Of course, as you can't use MSG_NOSIGNAL anymore, you'd also need to
```C
signal(SIGPIPE, SIG_IGN);
```
which is a good idea in any case.

I think for extensiblity, it would be nice to output some kind of JSON object instead of a plain number, similar to the fastd status socket. Maybe simply
```C
dprintf(fd, "{\"clients\":%zu}", count);
```
or something similar, depending on the way we plan to extend this format.